File: app/src/main/java/com/xudadong/leetcode/MainActivity.java
package com.xudadong.leetcode;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.xudadong.leetcode.adapter.MainAdapter;
import com.xudadong.leetcode.contract.Model;
import com.xudadong.leetcode.utils.ParseUtil;

import java.util.List;
import java.util.concurrent.Callable;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {

    private Disposable mDisposable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
    }

    private void initViews() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(R.string.app_name);
        }

        final RecyclerView vRecyclerView = findViewById(R.id.recyclerView);
        final TextView vLoading = findViewById(R.id.loading);
        vRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        vRecyclerView.setVisibility(View.INVISIBLE);
        vLoading.setVisibility(View.VISIBLE);

        mDisposable = Observable.defer((Callable<ObservableSource<List<Model>>>) () -> Observable.fromArray(ParseUtil.parseData()))
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(leetModels -> {
                    vRecyclerView.setVisibility(View.VISIBLE);
                    vLoading.setVisibility(View.INVISIBLE);

                    MainAdapter adapter = new MainAdapter(this, leetModels);
                    vRecyclerView.setAdapter(adapter);
                });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_about) {
            startActivity(new Intent(getApplicationContext(), AboutActivity.class));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mDisposable != null && !mDisposable.isDisposed()) {
            mDisposable.dispose();
            mDisposable = null;
        }
    }
}

File: app/src/main/java/com/xudadong/leetcode/MainActivity.kt
package com.xudadong.leetcode

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.xudadong.leetcode.adapter.MainAdapter
import com.xudadong.leetcode.ui.AboutActivity
import com.xudadong.leetcode.ui.BaseActivity
import com.xudadong.leetcode.utils.ParseUtil
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    private var mDisposable: Disposable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initActionBar(getString(R.string.app_name), false)
        initViews()
    }

    private fun initViews() {
        vRecyclerView.layoutManager = LinearLayoutManager(this)
        vRecyclerView.visibility = View.INVISIBLE
        vLoading.visibility = View.VISIBLE

        mDisposable = Observable.defer { Observable.fromArray(ParseUtil.parseData()) }
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe {
                vRecyclerView.visibility = View.VISIBLE
                vLoading.visibility = View.INVISIBLE

                val adapter = MainAdapter(this, it)
                vRecyclerView.adapter = adapter
            }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_about) {
            startActivity(Intent(applicationContext, AboutActivity::class.java))
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onDestroy() {
        if (mDisposable?.isDisposed == true) {
            mDisposable?.dispose()
            mDisposable = null
        }
        super.onDestroy()
    }
}

class Bird(name: String, private var age: Int) {

    init {
        this.age = 0
    }

    init {

    }
}